tests/workflows/test_snowflake.py

```diff
@@ -0,0 +1,110 @@
+        "account": os.environ["SNOWFLAKE_ACCOUNT"],
+        "warehouse": os.environ["SNOWFLAKE_WAREHOUSE"],
+        "role": os.environ.get("SNOWFLAKE_ROLE", "public"),
+        "database": os.environ.get("SNOWFLAKE_DATABASE") or "testdb",
+        "schema": os.environ.get("SNOWFLAKE_SCHEMA") or "testschema",
+    }
+
+
+@pytest.fixture
+def table(connection_kwargs):
+    """Connect to snowflake and create table"""
+    name = f"citibike_tripdata_{uuid.uuid4().hex}"
+    engine = create_engine(URL(**connection_kwargs))
+    engine.execute(f"DROP TABLE IF EXISTS {name}")
+    engine.execute(
+        f"""create table if not exists {name} (
+                ride_id varchar not null unique,
+                rideable_type varchar not null,
+                started_at timestamp not null,
+                ended_at timestamp not null,
+                start_station_name varchar not null,
+                start_station_id smallint not null,
+                end_station_name varchar not null,
+                end_station_id smallint not null,
+                start_lat number,
+                start_lng number,
+                end_lat number,
+                end_lng number,
+                is_member boolean not null
+            )"""
+    )
+    yield name
+    # after the data is written, delete table
+    engine.execute(f"DROP TABLE IF EXISTS {name}")
+
+
+@pytest.mark.client("snowflake")
+def test_write(client, connection_kwargs, table):
+    csv_paths = [
+        f"s3://tripdata/{ts.year}{ts.month:02}-*-*.csv.zip"
+        for ts in pd.date_range(start="2022-01-01", end="2023-03-01", freq="MS")
+    ]
+
+    # preprocess data
+    def safe_int(x):
+        """Some station IDs are not correct integers"""
+        try:
+            return int(float(x))
+        except Exception:
+            # if station ID is not an int, return -1
+            return -1
+
+    ddf = dd.read_csv(
+        csv_paths,
+        compression="zip",
+        blocksize=None,
+        converters={"start_station_id": safe_int, "end_station_id": safe_int},
+        storage_options={"anon": True},
+    )
+
+    # filter out incorrect station IDs
+    ddf = ddf[(ddf.start_station_id != -1) & (ddf.end_station_id != -1)].reset_index(
+        drop=True
+    )
+
+    # create boolean is_member and drop member_casual
+    ddf["is_member"] = ddf.member_casual == "member"
+
+    ddf = ddf.drop(columns="member_casual")
+
+    # repartition to ensure even chunks
+    ddf = ddf.repartition(partition_size="100Mb")
+
+    wait(ddf)
+
+    # save data to Snowflake
+    to_snowflake(ddf, name=table, connection_kwargs=connection_kwargs)
+
+
+@pytest.mark.client("snowflake")
+def test_read(client, connection_kwargs):
+    """Read and explore NYC bike dataset from Snowflake"""
+    table = "citibike_tripdata"  # persistent table
+
+    df = read_snowflake(
+        f"SELECT * FROM {table}",
+        connection_kwargs=connection_kwargs,
+        npartitions=200,
```

Comment:
I suspect users are more likely to specify `partition_size` than `npartitions` similar to the repartition in the write workload. Should we change (or simply remove this, from the docs it looks like we default to `100 MiB`).

Reply:
The default is None:
https://github.com/coiled/dask-snowflake/blob/fbfa369f78170015cfdf23cd1cc7918b1deb8fdf/dask_snowflake/core.py#L210
but it's a good point, most people would probably use `100MiB`. I'll change that.

Reply:
Sorry, what I meant was that we default within the function body:

https://github.com/coiled/dask-snowflake/blob/fbfa369f78170015cfdf23cd1cc7918b1deb8fdf/dask_snowflake/core.py#L263-L264